autoload/codefmt/mixformat.vim

```diff
@@ -0,0 +1,108 @@
+" Copyright 2022 Google Inc. All rights reserved.
+"
+" Licensed under the Apache License, Version 2.0 (the "License");
+" you may not use this file except in compliance with the License.
+" You may obtain a copy of the License at
+"
+"     http://www.apache.org/licenses/LICENSE-2.0
+"
+" Unless required by applicable law or agreed to in writing, software
+" distributed under the License is distributed on an "AS IS" BASIS,
+" WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+" See the License for the specific language governing permissions and
+" limitations under the License.
+
+let s:plugin = maktaba#plugin#Get('codefmt')
+let s:cmdAvailable = {}
+
+""
+" @private
+" Formatter: mixformat
+function! codefmt#mixformat#GetFormatter() abort
+  let l:formatter = {
+      \ 'name': 'mixformat',
+      \ 'setup_instructions': 'mix is usually installed with Elixir' .
+          \ '(https://elixir-lang.org/install.html). ' .
+          \ "If mix is not in your path, configure it in .vimrc:\n" .
+          \ 'Glaive codefmt mix_executable=/path/to/mix' }
+
+  function l:formatter.IsAvailable() abort
+    let l:cmd = codefmt#formatterhelpers#ResolveFlagToArray('mix_executable')
+    if codefmt#ShouldPerformIsAvailableChecks() && !executable(l:cmd[0])
+      return 0
+    endif
+    return 1
+  endfunction
+
+  function l:formatter.AppliesToBuffer() abort
+    return &filetype is# 'elixir' || &filetype is# 'eelixir'
+          \ || &filetype is# 'heex'
+  endfunction
+
+  ""
+  " Reformat the current buffer using ktfmt, only targeting {ranges}.
+  function l:formatter.FormatRange(startline, endline) abort
+    let l:filename = expand('%:p')
+    if empty(l:filename)
+      let l:filename = 'stdin.exs'
+    endif
+    " mix format docs: https://hexdocs.pm/mix/main/Mix.Tasks.Format.html
+    let l:cmd = codefmt#formatterhelpers#ResolveFlagToArray('mix_executable')
+    " Specify stdin as the file
+    let l:cmd = l:cmd + ['format', '--stdin-filename=' . l:filename, '-']
+    let l:dir = l:filename == 'stdin.exs' ? getcwd() : s:findMixDir(l:filename)
```

Comment:
Can stdin.exs be a valid filename here, or is it just supposed to be a placeholder value? If so, would this give the right behavior for a file explicitly named stdin.exs?

Reply:
Fair point.  I moved l:dir assignment into the `if empty` block and added a comment giving context to `stdin.exs`.